src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java

```diff
@@ -196,6 +208,30 @@ public StatisticsService(StatisticsRepository statisticsRepository) {
         return result;
     }
 
+    /**
+     * Gets a list of maps, each map describing an entry in the database. The map has the two keys "day" and "amount",
+     * which map to the date and the amount of the findings. This method handles the spanType WEEKS_ORDERED
+     *
+     * @param outcome A List<Map<String, Object>>, containing the content which should be refactored into an array
+     * @param result the array in which the converted outcome should be inserted
+     * @param endDate the endDate
+     * @return an array, containing the values for each bar in the graph
+     */
+    private Integer[] createResultArrayForQuarter(List<Map<String, Object>> outcome, Integer[] result, ZonedDateTime endDate) {
+        int week;
+        for (Map<String, Object> map : outcome) {
+            ZonedDateTime date = (ZonedDateTime) map.get("day");
+            Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : null;
```

Comment:
This should default to `0` instead of `null` since we can't add null:
`Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : 0;`

(When this is null it gives me a NullPointerException in line 228.)

Reply:
Good point, adapted